### PR TITLE
Add Trino 418 release notes

### LINKED ISSUE #17450: Release notes for 418

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 418 release. Fixes #17450 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 10 May 2023

* #17425 ✅ rn ✅ docs
* #17445 ✅ rn ✅ docs
* #17448 ✅ rn ✅ docs

## 11 May 2023

* #17443 ✅ rn ✅ docs
* #17441 ✅ rn ✅ docs
* #17381 ✅ rn ✅ docs
* #17200 ✅ rn ✅ docs
* #17456 ✅ rn ✅ docs
* #17318 ✅ rn ✅ docs
* #17442 ✅ rn ✅ docs
* #17332 ✅ rn ✅ docs
* #17431 ✅ rn ✅ docs
* #17347 ✅ rn ✅ docs

## 12 May 2023

* #17127 
* #17469 ✅ rn ✅ docs
* #16205 ✅ rn ❌ docs
* #17409 ✅ rn ✅ docs
* #17390 ✅ rn ✅ docs
* #17222 ✅ rn ✅ docs
* #17365 ✅ rn ✅ docs
* #17473 ✅ rn ✅ docs
* #17482 ✅ rn ✅ docs
* #17393 ✅ rn ✅ docs
* #17455 ✅ rn ✅ docs

## 14 May 2023

* #17495 ✅ rn ✅ docs

## 15 May 2023

* #17449 ❌ rn ✅ docs
* #17378 ✅ rn ✅ docs
* #17491 ✅ rn ✅ docs
* #17480 ✅ rn ✅ docs
* #17483 ✅ rn ✅ docs
* #17470 ✅ rn ✅ docs
* #17488 ✅ rn ✅ docs
* #17432 ✅ rn ✅ docs
* #17444 ❌ rn ✅ docs
* #17478 ✅ rn ✅ docs
* #17475 ✅ rn ✅ docs
* #17484 ✅ rn ✅ docs
* #17503 ✅ rn ✅ docs
* #16979 ✅ rn ✅ docs
* #17489 ✅ rn ✅ docs
* #16600 ✅ rn ✅ docs

## 16 May 2023

* #17493 ✅ rn ✅ docs
* #17435 ✅ rn ✅ docs
* #17417 ✅ rn ✅ docs
* #17323 ✅ rn ✅ docs
* #17506 ✅ rn ✅ docs
* #17465 ✅ rn ✅ docs
* #16187 ✅ rn ✅ docs (#16308)
* #17102 ✅ rn ✅ docs
* #17341 ✅ rn ✅ docs
* #17424 ✅ rn ✅ docs
* #17505 ✅ rn ✅ docs
* #17529 ✅ rn ✅ docs

## 17 May 2023

* #17434 ✅ rn ✅ docs
* #17533 ✅ rn ✅ docs
* #17534 ✅ rn ✅ docs
* #17535 ✅ rn ✅ docs
* #14493 ✅ rn ✅ docs
* #17479 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-418.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Release 418 (17 May 2023)
+
+## General
+
+* Add support for `EXECUTE IMMEDIATE`. ({issue}`17341`)
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:6]: What's a "partitioned column". Is this a reference to a Hive or Iceberg "partition column"? If so, it should go in the corresponding section(s). 

cc @raunaqmorarka

File: docs/src/main/sphinx/release/release-418.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## BigQuery connector
+
+* Add support for adding labels to BigQuery jobs started by Trino as part of
+  query processing. The name and value of the label can be configured via the
+  `bigquery.job.label-name` and `bigquery.job.label-format` catalog
+  configuration properties, respectively. ({issue}`16187`)
+
+## Delta Lake connector
+
+* Add support for DML statements for tables with an `id` column mapping. ({issue}`16600`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:18]: "DML statements" is not a thing. Expand to describe the actual functions that get affected.

File: docs/src/main/sphinx/release/release-418.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* Add `table_changes` table function. ({issue}`16205`)
+
+## Hive connector
+
+* Improve performance of querying `information_schema.tables` when using the
+  Hive metastore. ({issue}`17127`)
+* Enable the optimized Parquet writer by default. This can be disabled with
+  the `parquet.optimized-writer.enabled` configuration property or
+  the `parquet_optimized_writer_enabled` session property. ({issue}`17393`)
+* Share the value set by the
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:28]: What's the user-visible effect?

File: docs/src/main/sphinx/release/release-418.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  `hive.per-transaction-file-status-cache.max-retained-size` configuration
+  property across all running queries. ({issue}`17432`)
+* Remove the `temporary_staging_directory_enabled` and
+  `temporary_staging_directory_path` session property overrides for the
+  `temporary-staging-directory-enabled` and `temporary-staging-directory-path`
+  configuration properties. ({issue}`17390`)
+* Fix query failure for S3 objects with an empty `Content-Type` header. ({issue}`16546`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:35]: "empty `Content-Type` header" is a low level detail. Is there anything more abstract that describes the conditions under which the problem occurs?

Otherwise, I would just say something like:

"Fix failure when querying text files in S3 if the native reader is enabled".

File: docs/src/main/sphinx/release/release-418.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## Iceberg connector
+
+* Improve planning time for `SELECT` queries. ({issue}`17347`)
+* Fix incomplete results in the `$history` table when used with the REST
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:40]: > Fix incorrect results when querying the `$history` table if the REST catalog is enabled.

File: docs/src/main/sphinx/release/release-418.md (hunk 1 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-418.md:6]: Looks like it does reference partitioned columns from various connectors - I'll move it to Hive/Iceberg/Hudi/Delta, but are there any others?

Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-418.md]: Yes, it refers to a partition column in data lake connectors because that's the most straightforward case. The change is not in any connector though, it's in the engine join operator. So any connector which can produce RLE blocks for the engine can benefit, but that's a low level detail.

Comment (colebow) [reply docs/src/main/sphinx/release/release-418.md]: Yeah, I think we just want to frame it as where users will see this improvement. So data lake sections it is.

File: docs/src/main/sphinx/release/release-418.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+## General
+
+* Add support for `EXECUTE IMMEDIATE`. ({issue}`17341`)
+* Fix failure when invoking `current_timestamp`. ({issue}`17455`)
+
+## BigQuery connector
+
+* Add support for adding labels to BigQuery jobs started by Trino as part of
+  query processing. The name and value of the label can be configured via the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-418.md:11]: can we just link to the docs for this instead of adding the doc inline in the release notes?

File: docs/src/main/sphinx/release/release-418.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  `bigquery.job.label-name` and `bigquery.job.label-format` catalog
+  configuration properties, respectively. ({issue}`16187`)
+
+## Delta Lake connector
+
+* Add support for `INSERT`, `UPDATE`, `DELETE`, and `MERGE` statements for
+  tables with an `id` column mapping. ({issue}`16600`)
+* Add `table_changes` table function. ({issue}`16205`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-418.md:19]: Add the table_changes table function .. also do we need to explain a bit what its for or link to docs for it ?

File: docs/src/main/sphinx/release/release-418.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+
+## Hive connector
+
+* Improve performance of querying `information_schema.tables` when using the
+  Hive metastore. ({issue}`17127`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-418.md:25]: Is this only in hive connector or also the others?

File: docs/src/main/sphinx/release/release-418.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+* Improve performance of reading Parquet files by enabling the optimized Parquet 
+  writer by default. This can be disabled with the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-418.md:28]: The property is already documented .. I dont think we need to repeat that here

File: docs/src/main/sphinx/release/release-418.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  `parquet.optimized-writer.enabled` configuration property or the
+  `parquet_optimized_writer_enabled` session property. ({issue}`17393`)
+* Share the value set by the
+  `hive.per-transaction-file-status-cache.max-retained-size` configuration
+  property across all running queries. ({issue}`17432`)
+* Remove the `temporary_staging_directory_enabled` and
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-418.md:34]: Maybe just

```suggestion
* Remove the `temporary_staging_directory_enabled` and `temporary_staging_directory_path` session properties.
```

File: docs/src/main/sphinx/release/release-418.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+## Hudi connector
+
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+
+## Iceberg connector
+
+* Improve planning time for `SELECT` queries. ({issue}`17347`)
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+* Fix incorrect results when querying the `$history` table if the REST catalog
+  is enabled. ({issue}`17470`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-418.md:49]: is used.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-418.md (hunk 7 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-418.md:19]: Also not documented yet, so nothing to link to.

File: docs/src/main/sphinx/release/release-418.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
+# Release 418 (17 May 2023)
+
+## General
+
+* Add support for `EXECUTE IMMEDIATE`. ({issue}`17341`)
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-418.md:5]: rebase and change this to be a link

File: docs/src/main/sphinx/release/release-418.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+* Improve performance of reading Parquet files by enabling the optimized Parquet 
+  writer by default. ({issue}`17393`)
+* Remove the `temporary_staging_directory_enabled` and
+  `temporary_staging_directory_path` session properties. ({issue}`17390`)
+* Fix failure when querying text files in S3 if the native reader is enabled. ({issue}`16546`)
+
+## Hudi connector
+
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:35]: *partition* columns

File: docs/src/main/sphinx/release/release-418.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+  tables with an `id` column mapping. ({issue}`16600`)
+* Add the `table_changes` table function. ({issue}`16205`)
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+
+## Hive connector
+
+* Improve performance of querying `information_schema.tables` when using the
+  Hive metastore. ({issue}`17127`)
+* Improve performance of joins on partitioned columns. ({issue}`14493`)
+* Improve performance of reading Parquet files by enabling the optimized Parquet
```
Comment (martint) [inline docs/src/main/sphinx/release/release-418.md:27]: How is this related to improving performance of *reads*? Shouldn't this be "Improve performance of *writing* parquet data by enabling ..." ?

File: docs/src/main/sphinx/release/release-418.md (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-418.md:11]: Sorry .. wrong one..

File: docs/src/main/sphinx/release/release-418.md (hunk 8 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-418.md:25]: Looks like its for Hive views related aspects only.. and therefore the Hive connector only .. I think .. but also do we need to add the Hive views aspect?

**Review by mosabua — APPROVED** [review]

File: docs/src/main/sphinx/release/release-418.md (hunk 8 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-418.md:25]: Asked in code PR for more info

File: docs/src/main/sphinx/release/release-418.md (hunk 14 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-418.md]: It can potentially improve reads due to https://github.com/trinodb/trino/pull/17404 but the main benefit should be better memory usage accounting and speed of writer
